Packages/Ink/Editor/Core/Ink Library/InkFile.cs

```diff
@@ -7,87 +7,22 @@
-		public bool isMarkedToCompileAsMasterFile => InkSettings.instance.includeFilesToCompileAsMasterFiles.Contains(inkAsset);
-		public bool compileAutomatically => InkSettings.instance.filesToCompileAutomatically.Contains(inkAsset);
-		// A reference to the ink file
-		public DefaultAsset inkAsset;
-
-        //specify json destination folder (if None, default to same folder as ink file)
-        public DefaultAsset jsonAssetDirectory;
-
-		// The compiled json file. Use this to start a story.
-		public TextAsset jsonAsset;
-
-		// The file path relative to the Assets folder (Assets/Ink/Story.ink)
-		public string filePath {
-			get {
-				if(inkAsset == null) 
-					return null;
-
-				return InkEditorUtils.SanitizePathString(AssetDatabase.GetAssetPath(inkAsset));
-			}
-		}
-
-		// The full file path (C:/Users/Inkle/HeavensVault/Assets/Ink/Story.ink)
-		public string absoluteFilePath {
-			get {
-				if(inkAsset == null) 
-					return null;
-				return InkEditorUtils.UnityRelativeToAbsolutePath(filePath);
-			}
-		}
-
-		public string absoluteFolderPath => InkEditorUtils.SanitizePathString(Path.GetDirectoryName(absoluteFilePath));
-
-		// The path of any compiled json file. Relative to assets folder.
-        public string jsonPath {
-			get {
-                var _filePath = filePath;
-                Debug.Assert(!string.IsNullOrEmpty(_filePath), "File path for ink file is null! The ink library requires rebuilding. Asset: "+inkAsset);
-
-                DefaultAsset jsonFolder = jsonAssetDirectory;
-                if (jsonFolder == null) // no path specified for this specific file
-                {
-                    if(InkSettings.instance.defaultJsonAssetPath != null) 
-                    {
-                        // use default path in InkSettings
-                        jsonFolder = InkSettings.instance.defaultJsonAssetPath;
-                    }
-
-                    if (jsonFolder == null)
-                    {
-                        //fallback to same folder as .ink file
-                        jsonFolder = AssetDatabase.LoadAssetAtPath<DefaultAsset>(Path.GetDirectoryName(_filePath));
-                    }
-                }
-
-                Debug.Assert(jsonFolder != null, "JSON folder not found for ink file at path. File Path: "+_filePath+". Asset: "+inkAsset);
-
-                string _jsonPath = AssetDatabase.GetAssetPath(jsonFolder);
-                Debug.Assert(Directory.Exists(_jsonPath), "JSON folder path is not a directory! Json Path: "+_jsonPath+". Asset: "+inkAsset);
-                string strJsonAssetPath = InkEditorUtils.CombinePaths(_jsonPath, Path.GetFileNameWithoutExtension(_filePath)) + ".json";
-                return strJsonAssetPath;
-			}
-		}
-
-		public string absoluteJSONPath {
-			get {
-				if(inkAsset == null) 
-					return null;
-				return InkEditorUtils.UnityRelativeToAbsolutePath(jsonPath);
-			}
-		}
-
-
-
+	public class InkFile : ScriptableObject {
```

Comment:
I'm not familiar with the way the project is maintained but this is a breaking change and it would be nice to bump the major version in the package manifest with that.
```json
{
  "name": "com.inkle.ink-unity-integration",
  "version": "2.0.0",
  ...
}
```